### DISCUSSION TITLE
Peppol Directory Truststore Question

### DISCUSSION THREAD

**Discussion by canilsenlogiq** [phax/phoss-smp#211] [Q&A] [answered]
Hello.

We discovered that our **_Peppol Directory Truststore_** expires in 2 months. But we have no idea what it does.
It looks like we have an older truststore named prod-truststore.jks. Where we suspect that .jks have been downloaded from your truststore resource page (_https://github.com/phax/peppol-commons/tree/master/peppol-commons/src/main/resources/truststore_).

**Q:** Would updating our expiring truststore to your _**complete-truststore.jks**_ be correct/ok, or do you have any other suggestions?

Thank you very much for your attention.
Christoffer A. Nilsen

**Comment by phax** [comment, marked as answer]
Hi Christopher,
Thanks for reaching out to me.
The "Peppol Directory Truststore" is usually the same truststore that you use for the SMP itself. It needs to contain the public TLS certificates when building a TLS connection to the Peppol Directory to ask for an update of the Business Cards.

If you take the latest version of the "complete-truststore" from said source (https://github.com/phax/peppol-commons/tree/master/peppol-commons/src/main/resources/truststore) and just replace your existing one with that version, you should be on the safe side. I crosschecked, and in the latest version, the next expiration is in 2024.

Make sure to restart the SMP instance to take the new truststore.

hth, Philip
